### PR TITLE
Add deployment merge request api

### LINKED ISSUE #995: Add deployment merge request API
https://docs.gitlab.com/ee/api/deployments.html#list-of-merge-requests-associated-with-a-deployment

### PR DESCRIPTION
**Pull request by ludwig-weiss**
Hi,

I added the https://docs.gitlab.com/ee/api/deployments.html#list-of-merge-requests-associated-with-a-deployment API to `objects.py`.

Fixes #995 

Cheers
Ludwig

### CODE REVIEW

**Comment by max-wittig** [comment]
Hi @ludwig-weiss. Thanks for your MR! Would you mind to add a couple of tests and some docs?

**Comment by ludwig-weiss** [comment]
> Hi @ludwig-weiss. Thanks for your MR! Would you mind to add a couple of tests and some docs?

Not at all, but I was unsure what and how to test. Could you point me to an example?

**Comment by max-wittig** [comment]
I guess the tests are going through a little transition phase right now, regarding their structure. Sorry! But you can just add tests similar to this: https://github.com/python-gitlab/python-gitlab/blob/master/gitlab/tests/objects/test_projects.py. Thanks 👍

**Comment by ludwig-weiss** [comment]
I am not a python/python tests pro. The example seems too broad to me.

Is there a test for e.g ProjectPipelineJobManager https://github.com/python-gitlab/python-gitlab/blob/master/gitlab/v4/objects.py#L3651

This seems similar to the classes I added.
A manager with only ListMixin and RestManager added.

When I try to figure what to test for this GET only API I come to this:
 - mock one or more results of this api
 - call the code
 - compare the result with the mock

For this simple GET API I am under the impression that we would test the mock only.
Am I wrong here?

**Comment by max-wittig** [comment]
Yes it does test the mock only, but the functional tests alone are kind of unreliable.

**Comment by nejch** [comment]
@ludwig-weiss to avoid blocking this PR as the project structure has changed quite a bit and we're getting merge conflicts, I rebased your branch and added some docs and tests. You're still credited as the author of the commit, hope you're ok with this ;)

File: gitlab/v4/objects/deployments.py [hunk 1]
```diff
@@ -1,21 +1,23 @@
...
-    pass
+    _managers = (("mergerequests", "ProjectDeploymentMergeRequestManager"),)
 
 
 class ProjectDeploymentManager(RetrieveMixin, CreateMixin, UpdateMixin, RESTManager):
     _path = "/projects/%(project_id)s/deployments"
     _obj_cls = ProjectDeployment
     _from_parent_attrs = {"project_id": "id"}
-    _list_filters = ("order_by", "sort")
+    _list_filters = ("order_by", "sort", "environment", "status")
```
Comment (JohnVillalovos) [inline gitlab/v4/objects/deployments.py:20]: What about `updated_after` and `updated_before` ?

https://docs.gitlab.com/ee/api/deployments.html#list-project-deployments

File: gitlab/v4/objects/merge_requests.py [hunk 2]
```diff
@@ -409,6 +411,16 @@ class ProjectMergeRequestManager(CRUDMixin, RESTManager):
...
 
+class ProjectDeploymentMergeRequest(ProjectMergeRequest):
+    pass
+
+
+class ProjectDeploymentMergeRequestManager(ListMixin, RESTManager):
+    _path = "/projects/%(project_id)s/deployments/%(deployment_id)s/merge_requests"
+    _obj_cls = ProjectDeploymentMergeRequest
+    _from_parent_attrs = {"deployment_id": "id", "project_id": "project_id"}
+
```
Comment (JohnVillalovos) [inline gitlab/v4/objects/merge_requests.py:422]: https://docs.gitlab.com/ee/api/deployments.html#list-of-merge-requests-associated-with-a-deployment

> It supports the same parameters as the Merge Requests API and returns a response using the same format: 

So should this class be derived from a MergeRequest class?

Comment (nejch) [reply gitlab/v4/objects/merge_requests.py:432]: Makes sense, changed. CLI when inheriting from MergeRequestManager:
```shell
$ gitlab project-deployment-merge-request list --help
usage: gitlab project-deployment-merge-request list [-h] [--sudo SUDO] --deployment-id DEPLOYMENT_ID --project-id PROJECT_ID [--state STATE] [--order-by ORDER_BY]
                                                    [--sort SORT] [--milestone MILESTONE] [--view VIEW] [--labels LABELS] [--with-labels-details WITH_LABELS_DETAILS]
                                                    [--with-merge-status-recheck WITH_MERGE_STATUS_RECHECK] [--created-after CREATED_AFTER] [--created-before CREATED_BEFORE]
                                                    [--updated-after UPDATED_AFTER] [--updated-before UPDATED_BEFORE] [--scope SCOPE] [--author-id AUTHOR_ID]
                                                    [--author-username AUTHOR_USERNAME] [--assignee-id ASSIGNEE_ID] [--approver-ids APPROVER_IDS]
                                                    [--approved-by-ids APPROVED_BY_IDS] [--reviewer-id REVIEWER_ID] [--reviewer-username REVIEWER_USERNAME]
                                                    [--my-reaction-emoji MY_REACTION_EMOJI] [--source-branch SOURCE_BRANCH] [--target-branch TARGET_BRANCH] [--search SEARCH]
                                                    [--in IN] [--wip WIP] [--not NOT] [--environment ENVIRONMENT] [--deployed-before DEPLOYED_BEFORE]
                                                    [--deployed-after DEPLOYED_AFTER] [--page PAGE] [--per-page PER_PAGE] [--all]
```

File: gitlab/v4/objects/deployments.py (hunk 1 above)
Comment (nejch) [reply gitlab/v4/objects/deployments.py]: Actually not sure why this was part of the original PR, but I've added the latest params.

**Review by JohnVillalovos — APPROVED** [review]

**Pull request merged**